x-pack/plugins/ml/public/application/explorer/dashboard_controls/add_anomaly_charts_to_dashboard_controls.tsx

```diff
@@ -0,0 +1,102 @@
+import { useDashboardTable } from './use_dashboards_table';
+import { AddToDashboardControl } from './add_to_dashboard_controls';
+import { useAddToDashboardActions } from './use_add_to_dashboard_actions';
+import { AppStateSelectedCells, getSelectionTimeRange } from '../explorer_utils';
+import { TimeRange } from '../../../../../../../src/plugins/data/common/query';
+import { DEFAULT_MAX_SERIES_TO_PLOT } from '../../services/anomaly_explorer_charts_service';
+import { JobId } from '../../../../common/types/anomaly_detection_jobs';
+import { ANOMALY_EXPLORER_CHARTS_EMBEDDABLE_TYPE } from '../../../embeddables';
+import { getDefaultExplorerChartsPanelTitle } from '../../../embeddables/anomaly_charts/anomaly_charts_embeddable';
+import { TimeRangeBounds } from '../../util/time_buckets';
+import { useTableSeverity } from '../../components/controls/select_severity';
+
+function getDefaultEmbeddablePanelConfig(jobIds: JobId[]) {
+  return {
+    type: ANOMALY_EXPLORER_CHARTS_EMBEDDABLE_TYPE,
+    title: getDefaultExplorerChartsPanelTitle(jobIds),
+  };
+}
+
+export interface AddToDashboardControlProps {
+  jobIds: string[];
+  selectedCells?: AppStateSelectedCells;
+  bounds?: TimeRangeBounds;
+  interval?: number;
+  onClose: (callback?: () => Promise<any>) => void;
+}
+
+/**
+ * Component for attaching anomaly swim lane embeddable to dashboards.
+ */
+export const AddAnomalyChartsToDashboardControl: FC<AddToDashboardControlProps> = ({
+  onClose,
+  jobIds,
+  selectedCells,
+  bounds,
+  interval,
+}) => {
+  const [severity] = useTableSeverity();
+
+  const getPanelsData = useCallback(async () => {
+    let timeRange: TimeRange | undefined;
+    if (selectedCells !== undefined && interval !== undefined && bounds !== undefined) {
+      const { earliestMs, latestMs } = getSelectionTimeRange(selectedCells, interval, bounds);
+      timeRange = {
+        from: formatDate(earliestMs, 'MMM D, YYYY @ HH:mm:ss.SSS'),
+        to: formatDate(latestMs, 'MMM D, YYYY @ HH:mm:ss.SSS'),
+        mode: 'absolute',
+      };
+    }
+
+    const config = getDefaultEmbeddablePanelConfig(jobIds);
+    return [
+      {
+        ...config,
+        embeddableConfig: {
+          jobIds,
+          maxSeriesToPlot: DEFAULT_MAX_SERIES_TO_PLOT,
+          severityThreshold: severity.val,
+          ...(timeRange ?? {}),
+        },
+      },
+    ];
+  }, [selectedCells, interval, bounds]);
+
+  const { selectedItems, selection, dashboardItems, isLoading, search } = useDashboardTable();
+  const { addToDashboardAndEditCallback, addToDashboardCallback } = useAddToDashboardActions({
+    onClose,
+    getPanelsData,
+    selectedDashboards: selectedItems,
+  });
+  const title = (
+    <FormattedMessage
+      id="xpack.ml.explorer.anomalies.dashboardsTitle"
+      defaultMessage="Add anomaly charts to dashboards"
+    />
+  );
+
+  const disabled = selectedItems.length < 1 && !Array.isArray(jobIds === undefined);
+  return (
+    <AddToDashboardControl
```

Comment:
Might be nice to add the control allowing the user to set the maximum number of charts to this modal:

![image](https://user-images.githubusercontent.com/7405507/112870900-9dc03880-90b6-11eb-8597-3a0939cbbe00.png)

This would make it consistent with the behavior for the swim lane embeddable, where the user gets to select the type of swim lane to add (Overall / View by) before it gets added to the dashboard. Otherwise they have to go to the dashboard and edit the embedded charts there to alter the max series to plot.

Reply:
Updated here [`d71da6f` (#95623)](https://github.com/elastic/kibana/pull/95623/commits/d71da6fcc45d707b5954d6b0ed75d57c7fb00f74)